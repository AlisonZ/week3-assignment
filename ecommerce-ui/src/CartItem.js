import React, {Component} from 'react';
import PropTypes from 'prop-types';

class CartItem extends Component {
    static propTypes = {
        rental: PropTypes.object,
    }

    render() {
        return (
            <div className="cart-item">
                <span className="dark-grey-primary-title cart-title cart-details">{this.props.rental.title}</span>
                <img src={this.props.rental.image} className="cart-image cart-details" alt="rental-property"/>
                <span className="dark-grey-terciary cart-cost cart-details">
                    <div>
                        ${this.props.rental.payment.cost}/
                    </div>
                    <div>night</div>
                </span>
            </div>
        );
    }
}

export default CartItem;