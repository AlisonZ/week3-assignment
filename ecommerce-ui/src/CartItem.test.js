import React from 'react';
import {shallow} from 'enzyme';
import CartItem from './CartItem';
import { MOCK_RENTAL_1 } from "./TestMocks.js";

it('shallow renders Cart Item', () => {
    const wrapper = shallow(
    <CartItem rental={MOCK_RENTAL_1}/>
    );

    expect(wrapper).toMatchSnapshot();
});