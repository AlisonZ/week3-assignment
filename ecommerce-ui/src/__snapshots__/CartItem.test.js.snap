// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders Cart Item 1`] = `
<div
  className="cart-item"
>
  <span
    className="dark-grey-primary-title cart-title cart-details"
  >
    Centrally-located Manhattan studio
  </span>
  <img
    alt="rental-property"
    className="cart-image cart-details"
    src="https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80"
  />
  <span
    className="dark-grey-terciary cart-cost cart-details"
  >
    <div>
      $
      105
      /
    </div>
    <div>
      night
    </div>
  </span>
</div>
`;
